python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -1010,22 +1021,22 @@ def _construct_asset_record_from_row(
         else:
             check.failed("Row did not contain asset key.")
 
-    def _get_latest_materializations(
+    def _get_latest_materialization_records(
         self, raw_asset_rows
-    ) -> Mapping[AssetKey, Optional[EventLogEntry]]:
+    ) -> Mapping[AssetKey, Optional[EventLogRecord]]:
         # Given a list of raw asset rows, returns a mapping of asset key to latest asset materialization
         # event log entry. Fetches backcompat EventLogEntry records when the last_materialization
         # in the raw asset row is an AssetMaterialization.
         to_backcompat_fetch = set()
-        results: Dict[AssetKey, Optional[EventLogEntry]] = {}
+        results: Dict[AssetKey, Optional[EventLogRecord]] = {}
         for row in raw_asset_rows:
             asset_key = AssetKey.from_db_string(row[1])
             if not asset_key:
                 continue
             event_or_materialization = (
                 deserialize_json_to_dagster_namedtuple(row[2]) if row[2] else None
             )
-            if isinstance(event_or_materialization, EventLogEntry):
+            if isinstance(event_or_materialization, EventLogRecord):
```

Comment:
I think if the object is an event log entry we don't need to do the backcompat fetch to refetch the event. I think we can just only fetch the storage ID and combine it with the event log entry from the asset row

Reply:
This is definitely true, I just figured that this would add a lot more surface area to this change -- i'd have to keep track of two separate sets of things (the ones with just AssetMaterializations, and the ones with EventLogEntries), then have two separate backcompat queries. Just seemed like a lot of opportunity for error / more code to maintain, but interested if you have a different take